extensions/mqtt/processors/ConsumeMQTT.h

```diff
@@ -68,63 +77,82 @@ class ConsumeMQTT : public processors::AbstractMQTTProcessor {
-    processor->onSubscriptionFailure(response);
-  }
+  std::queue<SmartMessage> getReceivedMqttMessages();
+
+  // MQTT static async callbacks, calling their non-static counterparts with context being pointer to "this"
+  static void subscriptionSuccess(void* context, MQTTAsync_successData* response);
+  static void subscriptionSuccess5(void* context, MQTTAsync_successData5* response);
+  static void subscriptionFailure(void* context, MQTTAsync_failureData* response);
+  static void subscriptionFailure5(void* context, MQTTAsync_failureData5* response);
 
-  void onSubscriptionSuccess(MQTTAsync_successData* /*response*/) {
-    logger_->log_info("Successfully subscribed to MQTT topic %s on broker %s", topic_, uri_);
+  // MQTT non-static async callbacks
+  void onSubscriptionSuccess();
+  void onSubscriptionFailure(MQTTAsync_failureData* response);
+  void onSubscriptionFailure5(MQTTAsync_failureData5* response);
+  void onMessageReceived(SmartMessage smart_message) override;
+
+  void enqueueReceivedMQTTMsg(SmartMessage message);
+  void startupClient() override;
+  void checkProperties() override;
+  void checkBrokerLimitsImpl() override;
+
+  void resolveTopicFromAlias(const std::unique_ptr<MQTTAsync_message, MQTTMessageDeleter>& message, std::string& topic);
+
+  bool getCleanSession() const override {
+    return clean_session_;
   }
 
-  void onSubscriptionFailure(MQTTAsync_failureData* response) {
-    logger_->log_error("Subscription failed on topic %s to MQTT broker %s (%d)", topic_, uri_, response->code);
-    if (response->message != nullptr) {
-      logger_->log_error("Detailed reason for subscription failure: %s", response->message);
-    }
+  bool getCleanStart() const override {
+    return clean_start_;
   }
 
-  bool getCleanSession() const override {
-    return cleanSession_;
+  std::chrono::seconds getSessionExpiryInterval() const override {
+    return session_expiry_interval_;
   }
 
-  void onMessageReceived(char* topic_name, int /*topic_len*/, MQTTAsync_message* message) override;
+  void putUserPropertiesAsAttributes(const SmartMessage& message, const std::shared_ptr<core::FlowFile>& flow_file, const std::shared_ptr<core::ProcessSession>& session) const;
+  void fillAttributeFromContentType(const SmartMessage& message, const std::shared_ptr<core::FlowFile>& flow_file, const std::shared_ptr<core::ProcessSession>& session) const;
 
-  void enqueueReceivedMQTTMsg(std::unique_ptr<MQTTAsync_message, MQTTMessageDeleter> message);
+  void setMqtt5ConnectOptionsImpl(MQTTProperties& connect_props) const override;
 
-  bool startupClient() override;
+  std::string topic_;
+  bool clean_session_ = true;
+  bool clean_start_ = true;
+  std::chrono::seconds session_expiry_interval_{0};
+  uint64_t max_queue_size_ = 1000;
+  std::string attribute_from_content_type_;
 
-  void checkProperties() override;
+  uint16_t topic_alias_maximum_{0};
+  uint16_t receive_maximum_{65535};
```

Comment:
this seems to be `std::numeric_limits<uint16_t>::max()`

Reply:
this number is repeated in 5 places (in mqtt related files), should we introduce a MQTT_RECEIVE_MAX or something similar?

Reply:
I called it `MQTT_MAX_RECEIVE_MAXIMUM`, because this is the maximum possible value for the Receive Maximum MQTT property. This could either be defined by the client or the broker and it means the maximum number of unacknowledged messages they can tolerate at a time.